client/me/purchases/payment/edit-card-details/index.jsx

```diff
@@ -86,6 +92,10 @@ const EditCardDetails = React.createClass( {
 		this.setState( { form: this.formStateController.getInitialState() } );
 	},
 
+	componentWillReceiveProps( nextProps ) {
+		this.redirectIfDataIsInvalid( nextProps );
```

Comment:
Good catch. I opened an issue for this here: https://github.com/Automattic/wp-calypso/issues/4590